RabbitMQ.xs

```diff
@@ -41,16 +41,20 @@
 
 typedef amqp_connection_state_t Net__AMQP__RabbitMQ;
 
-/* this is a place to put some stuff that we convert from perl, it's transient and we recycle it as soon as it's finished being used, which means we keep memory we've used with the aim of reusing it */
-amqp_pool_t temp_memory_pool;
-
-//mashup of things to free memory, also temp_memory_pool is ugly and code smell
-void maybe_recycle_memory(amqp_connection_state_t conn)
-{
-    if (amqp_release_buffers_ok(conn)) {
-        amqp_release_buffers(conn);
-        recycle_amqp_pool( &temp_memory_pool );
-    }
+#define AMQP_STATUS_UNKNOWN_TYPE 0x500
+
+/* This is a place to put some stuff that we convert from perl,
+   it's transient and we recycle it as soon as it's finished being used
+   which means we keep memory we've used with the aim of reusing it */
+/* temp_memory_pool is ugly and suffers from code smell */
+static amqp_pool_t temp_memory_pool;
```

Comment:
Besides the fact that they should be static on principle -- a fundamental tenet of Computer Science is to limit access -- name collisions with other XS modules are possibles when using static linking.